### ISSUE TITLE
JDBC error when connecting to SQLite db

### ISSUE THREAD

**Issue by pdagosto** [schemacrawler/SchemaCrawler#394]
# Bug Report

**Issue**

Please explain the issue briefly
CentOS-7
schemacrawler-16.11.6
java 1.8.0_171
jdbc:  sqlite-jdbc-3.32.3.2.jar (included in distribution)

I  downloaded version v16.11.6 (schemacrawler-16.11.6-distribution.zip and tried the sqlite example:

cd /work/schemacrawler-16.11.6-distribution/examples/diagram/

and got this result:

./sqlite_diagram.sh /work/repos/bfx-workflow/workflows.db workflows_db.pdf
Failed to load native library:sqlite-3.32.3.2-7daaa90f-2e20-4c2b-8c0c-c351e3b842e9-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: /tmp/sqlite-3.32.3.2-7daaa90f-2e20-4c2b-8c0c-c351e3b842e9-libsqlitejdbc.so: /tmp/sqlite-3.32.3.2-7daaa90f-2e20-4c2b-8c0c-c351e3b842e9-libsqlitejdbc.so: failed to map segment from shared object: Operation not permitted
SchemaCrawler 16.11.6

Error: Could not connect to jdbc:sqlite:/work/repos/bfx-workflow/workflows.db, for user 'sa', with properties {}: Error opening connection

I also copied the command line from the example script and ran it from the command line using -loglevel=INFO:

../../_schemacrawler/schemacrawler.sh --server=sqlite --database=/work/repos/bfx-workflow/backend/workflows.db --user=sa --password= --info-level=maximum -c=schema --output-format=pdf -o=workflows.pdf -loglevel=ALL > schemacrawler.log 2>&1

The contents of the log are:

Failed to load native library:sqlite-3.32.3.2-ff6d572d-5d93-43d0-99d9-43ded60745ab-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: /tmp/sqlite-3.32.3.2-ff6d572d-5d93-43d0-99d9-43ded60745ab-libsqlitejdbc.so: /tmp/sqlite-3.32.3.2-ff6d572d-5d93-43d0-99d9-43ded60745ab-libsqlitejdbc.so: failed to map segment from shared object: Operation not permitted
SchemaCrawler 16.11.6

Error: Could not connect to jdbc:sqlite:/work/repos/bfx-workflow/backend/workflows.db, for user 'sa', with properties {}: Error opening connection

Run SchemaCrawler again with just the
-h
option for help

Or, run SchemaCrawler again with an additional
--log-level=CONFIG
option for details on the error

Additional resources are on the SchemaCrawler website,
https://www.schemacrawler.com
and a live, interactive tutorial is on
https://katacoda.com/schemacrawler

**Comment by sualeh** [comment]
@pdagosto - this error is coming from the [xerial/sqlite-jdbc](https://github.com/xerial/sqlite-jdbc) driver. I tried searching for the error message, but could not find someone who has encountered this. You can try with the latest version of the driver. Remove the old jar file from the SchemaCrawler `lib/` folder, and copy the latest one in. If that does not work, you may have to open an issue with the [xerial/sqlite-jdbc](https://github.com/xerial/sqlite-jdbc) project.

**Comment by pdagosto** [comment]
I've investigated further and found that the error was actually:

java.lang.UnsatisfiedLinkError: /tmp/sqlite-3.32.3.2-9c96fb8b-8402-43e1-bfd9-9974fbd99a84-libsqlitejdbc.so: /tmp/sqlite-3.32.3.2-9c96fb8b-8402-43e1-bfd9-9974fbd99a84-libsqlitejdbc.so: failed to map segment from shared object: Operation not permitted

which, after still further investigation is a matter of the /tmp directory being mounted with the noexec flag. Unfortunately I am not at liberty to change to mount properties of the /tmp directory (controlled by IT) so I'm not sure how to work around this. I've tried putting export JAVA_OPTS=-Djava.io.tmpdir=_some_other_directory_ in my .bashrc but  after sourcing my .bashrc file I still got the same error when I ran schemacrawler.

I am cross-filing this on xerial's GitHub site.
